File: exercise/kotlin/week06-exercise2/src/main/kotlin/report/ReportGenerator.kt
package report

class ReportGenerator {
    fun generateReport(reportType: String, data: List<ReportData>) {
        when (reportType) {
            "CSV" -> {
                println("Starting CSV Report Generation...")
                println("CSV Header: ID, Value, Description")
                for (d in data) {
                    println("${d.id},${d.value},${d.description}")
                }
                println("CSV Report Generated Successfully.")
            }
            "PDF" -> {
                println("Starting PDF Report Generation...")
                println("PDF Report Title: Comprehensive Data Report")
                println("------------------------------------------------")
                for (d in data) {
                    println("Data ID: ${d.id} | Data Value: ${d.value} | Description: ${d.description}")
                }
                println("------------------------------------------------")
                println("PDF Report Generated Successfully.")
            }
            else -> println("Report type $reportType not supported.")
        }
    }
}

fun main() {
    val generator = ReportGenerator()
    val data = listOf(
        ReportData(1, 100.0, "Sample Data 1"),
        ReportData(2, 200.0, "Sample Data 2")
    )
    generator.generateReport("CSV", data)
    generator.generateReport("PDF", data)
}

File: exercise/c#/week06-legacy-code/Exercise2ReportGenerator/ReportGenerator.cs
namespace Exercise2ReportGenerator
{
    class ReportGenerator
    {
        public void GenerateReport(string reportType, List<ReportData> data)
        {
            if (reportType == "CSV")
            {
                Console.WriteLine("Starting CSV Report Generation...");
                Console.WriteLine("CSV Header: ID, Value, Description");
                foreach (var d in data)
                {
                    Console.WriteLine($"{d.getId()},{d.getValue()},{d.getDescription()}");
                }
                Console.WriteLine("CSV Report Generated Successfully.");
            }
            else if (reportType == "PDF")
            {
                Console.WriteLine("Starting PDF Report Generation...");
                Console.WriteLine("PDF Report Title: Comprehensive Data Report");
                Console.WriteLine("------------------------------------------------");
                foreach (var d in data)
                {
                    Console.WriteLine($"Data ID: {d.getId()} | Data Value: {d.getValue()} | Description: {d.getDescription()}");
                }
                Console.WriteLine("------------------------------------------------");
                Console.WriteLine("PDF Report Generated Successfully.");
            }
            else
            {
                Console.WriteLine($"Report type {reportType} not supported.");
            }
        }
    }
}